aggregator/entry.go

```diff
@@ -118,8 +172,8 @@ func (e *Entry) AddMetricWithPolicies(
 		return errEntryClosed
 	}
 
-	if e.shouldUpdatePoliciesWithLock(currTime, policies.Version, policies.Cutover) {
-		if err := e.updatePoliciesWithLock(mu.Type, mu.ID, mu.OwnsID, policies.Version, policies.Policies()); err != nil {
+	if e.shouldUpdatePoliciesWithLock(currTime, sp) {
```

Comment:
Yeah so this is doing double-checking in that we acquire a read lock first, and if the entry should be updated, we release the read lock and acquire the write lock, but in between lock promotion, `e.closed` might change, so we need to check it again.